src/site/SearchResultSet.php

```diff
@@ -0,0 +1,52 @@
+<?hh // strict
+
+namespace HHVM\UserDocumentation;
+
+final class SearchResultSet {
+    private Map<string, string> $classes = Map {};
+    private Map<string, string> $functions = Map {};
+    private Map<string, string> $interfaces = Map {};
+    private Map<string, string> $hhvm_guides = Map {};
+    private Map<string, string> $hack_guides = Map {};
+
+    public function addAPIResult(string $type, string $name): void {
```

Comment:
Use APIDefinitionType enum (new) and switch statements (so that the typechecker makes sure that you handle all the enum values)